test/dlc_tests/conftest.py

```diff
@@ -222,10 +235,112 @@ def pytest_runtest_setup(item):
             pytest.skip("Skipping non-canary tests")
 
 
+def pytest_collection_modifyitems(session, config, items):
+    if config.getoption("--generate-coverage-doc"):
+        test_coverage_file = test_utils.TEST_COVERAGE_FILE
+        test_cov = {}
+        for item in items:
+            # Define additional csv options
+            function_name = item.name.split("[")[0]
+            function_key = f"{item.fspath}::{function_name}"
+            str_fspath = str(item.fspath)
+            str_keywords = str(item.keywords)
+
+            # Based on keywords and filepaths, assign values
+            scope = infer_field_value("all", ("mxnet", "tensorflow", "pytorch"), str_fspath)
+            train_inf = infer_field_value("both", ("training", "inference"), str_fspath, str_keywords)
+            integration = infer_field_value("general integration",
+                                            ("_dgl_", "smdebug", "gluonnlp", "smexperiments", "_mme_", "pipemode",
+                                             "tensorboard", "_s3_"),
+                                            str_keywords)
+            model = infer_field_value("N/A",
+                                      ("mnist", "densenet", "squeezenet", "half_plus_two", "half_plus_three"),
+                                      str_keywords)
+            num_instances = infer_field_value(1, ("_multinode_", "_multi-node_"), str_fspath, str_keywords)
+            cpu_gpu = infer_field_value("all", ("cpu", "gpu", "eia"), str_keywords)
+            if cpu_gpu == "gpu":
+                if "p2.xlarge" in str_keywords:
```

Comment:
p3.2xlarge is also single-GPU. Could you check other instance types [g*] that might be single-GPU?

Reply:
I will do this via boto3 apis